### PR TITLE
check shutdown/pause when in mid cast and interrupt.

### PR DESCRIPTION
**Pull request by RekkasGit**
move pause higher up
check loot cancel and pause/interrupt during looting

### CODE REVIEW

File: E3Next/Utility/e3util.cs [hunk 1]
```diff
@@ -516,6 +516,14 @@ public static void GiveItemOnCursorToTarget(bool moveBackToOriginalLocation = tr
                 e3util.TryMoveToLoc(currentX, currentY);
             }
         }
+        public static bool IsShuttingDown()
+        {
+            if(EventProcessor.CommandList["/shutdown"].queuedEvents.Count > 0)
```
Comment (jonnyrocket0) [inline E3Next/Utility/e3util.cs:521]: is this dict guaranteed to have this key?

File: E3Next/Processors/NowCast.cs [hunk 2]
```diff
@@ -95,6 +95,14 @@ private static void RegisterEvents()
 
         }
 
+        public static bool IsNowCastInQueue()
+        {
+            if(EventProcessor.CommandList["/nowcast"].queuedEvents.Count > 0)
```
Comment (jonnyrocket0) [inline E3Next/Processors/NowCast.cs:100]: is this dict guaranteed to have this key?

File: E3Next/Processors/Basics.cs [hunk 3]
```diff
@@ -695,10 +695,13 @@ public static void CheckManaResources()
 
                 bool hasManaStone = MQ.Query<bool>("${Bool[${FindItem[=Manastone]}]}");
                 bool amIStanding = MQ.Query<bool>("${Me.Standing}");
-                if (MQ.Query<bool>("${Me.Invis}")) return;
+               
 
                 if (hasManaStone && amIStanding)
                 {
+                    MQ.Delay(0);
```
Comment (jonnyrocket0) [inline E3Next/Processors/Basics.cs:702]: what does a delay(0) even do?

File: E3Next/Utility/e3util.cs (hunk 1 above)
Comment (RekkasGit) [reply E3Next/Utility/e3util.cs:521]: yes

File: E3Next/Processors/NowCast.cs (hunk 2 above)
Comment (RekkasGit) [reply E3Next/Processors/NowCast.cs:100]: yes

File: E3Next/Processors/Basics.cs (hunk 3 above)
Comment (RekkasGit) [reply E3Next/Processors/Basics.cs:702]: gives control back to MQ/EQ, so a new message can be pushed into the queues.

Comment (RekkasGit) [reply E3Next/Processors/Basics.cs:702]: I thought about making it a method just to be more descriptive.

**Review by jonnyrocket0 — APPROVED** [review]

File: E3Next/Processors/Basics.cs (hunk 3 above)
Comment (jonnyrocket0) [reply E3Next/Processors/Basics.cs:702]: I think that makes sense; something like YieldToEq() or something

**Pull request merged**